src/elm/Logs.elm

```diff
@@ -111,25 +113,55 @@ viewLogs stepNumber logFocus log clickAction shiftDown =
 -}
 viewLines : StepNumber -> LogFocus -> Maybe (WebData Log) -> SetLogFocus msg -> Bool -> Html msg
 viewLines stepNumber logFocus log clickAction shiftDown =
-    div [ class "lines" ] <|
-        List.indexedMap
-            (\idx -> \line -> viewLine stepNumber line logFocus (idx + 1) clickAction shiftDown)
-        <|
-            decodeLogLine log
-
-
-{-| viewLine : takes step number, line focus information, and click action and renders a log line
+    Html.table [ class "log-table" ] <|
+        Array.toList <|
+            Array.indexedMap
+                (\idx line ->
+                    viewLine
+                        { id = stepNumber
+                        , lineNo = idx + 1
+                        , line = line
+                        , stepNumber = stepNumber
+                        , logFocus = logFocus
+                        , setLogFocus = clickAction
+                        , shiftDown = shiftDown
+                        }
+                )
+            <|
+                decodeAnsi log
+
+
+{-| viewLine : takes log line and focus information and renders line number button and log
 -}
-viewLine : StepNumber -> String -> LogFocus -> Int -> SetLogFocus msg -> Bool -> Html msg
-viewLine stepNumber line logFocus lineNumber clickAction shiftDown =
-    div [ class "line" ]
+viewLine :
+    { id : String
+    , lineNo : Int
+    , line : Ansi.Log.Line
+    , stepNumber : StepNumber
+    , logFocus : LogFocus
+    , setLogFocus : SetLogFocus msg
+    , shiftDown : Bool
+    }
+    -> Html msg
+viewLine { id, lineNo, line, stepNumber, logFocus, setLogFocus, shiftDown } =
+    Html.tr
+        [ Html.Attributes.id <|
+            id
+                ++ ":"
+                ++ String.fromInt lineNo
+        , class "line"
+        ]
         [ div
-            [ Util.testAttribute <| "log-line-" ++ String.fromInt lineNumber
-            , class "wrapper"
-            , logFocusStyles logFocus lineNumber
+            [ class "wrapper"
+            , Util.testAttribute <| String.join "-" [ "log", "line", stepNumber, String.fromInt lineNo ]
+            , logFocusStyles logFocus lineNo
             ]
-            [ lineFocusButton stepNumber logFocus lineNumber clickAction shiftDown
-            , code [] [ text line ]
+            [ Html.td []
+                [ lineFocusButton stepNumber logFocus lineNo setLogFocus shiftDown ]
+            , Html.td [ class "-word-break-all", class "-overflow-auto" ]
+                [ code [ Util.testAttribute <| String.join "-" [ "log", "data", stepNumber, String.fromInt lineNo ] ]
+                    [ Ansi.Log.viewLine line ]
```

Comment:
this is where you take a decoded ansi log line and render it with the appropriate style